integration_test/Makefile

```diff
@@ -199,9 +198,31 @@ pg13:
 	jq '.queryInformations[].normalizedQuery' pg13_2.json | sort --version-sort -f > pg13.out
 	diff -Naur pg13.expected pg13.out && echo 'success'
 
+# Test reload both with an empty config and a server present
+#
+# Note that we reload twice after adding a server to the config, so we can test the
+# "write state file when reload occurs" logic, that requires the server to be there
+# *before* the reload
+reload:
+	docker build -f Dockerfile.test-reload $(DOCKER_BUILD_OPTS)
+	$(call docker_run_cmd,)
+	docker exec --user pganalyze --detach pganalyze-collector-test bash -c "pganalyze-collector 2> /tmp/collector.log"
+	sleep 5
+	docker exec pganalyze-collector-test pganalyze-collector --reload
+	docker exec pganalyze-collector-test cat /tmp/collector.log | grep "Reloading configuration..."
```

Comment:
I think this could technically be racy (since I think the signal is processed asynchronously?), but I don't think we don't need to worry about it.

Reply:
Yeah, good point - I think its fine, but agree that there is a theoretical race condition here.